naga/src/proc/constant_evaluator.rs

```diff
@@ -592,186 +771,82 @@ impl<'a> ConstantEvaluator<'a> {
         }
 
         match fun {
-            crate::MathFunction::Pow => self.math_pow(arg, arg1.unwrap(), span),
-            crate::MathFunction::Clamp => self.math_clamp(arg, arg1.unwrap(), arg2.unwrap(), span),
-            fun => Err(ConstantEvaluatorError::NotImplemented(format!(
-                "{fun:?} built-in function"
-            ))),
-        }
-    }
-
-    fn math_pow(
-        &mut self,
-        e1: Handle<Expression>,
-        e2: Handle<Expression>,
-        span: Span,
-    ) -> Result<Handle<Expression>, ConstantEvaluatorError> {
-        let e1 = self.eval_zero_value_and_splat(e1, span)?;
-        let e2 = self.eval_zero_value_and_splat(e2, span)?;
-
-        let expr = match (&self.expressions[e1], &self.expressions[e2]) {
-            (&Expression::Literal(Literal::F32(a)), &Expression::Literal(Literal::F32(b))) => {
-                Expression::Literal(Literal::F32(a.powf(b)))
+            crate::MathFunction::Abs => {
+                component_wise_scalar(self, span, [arg], |args| match args {
+                    Scalar::AbstractFloat([e]) => e.abs().into(),
+                    Scalar::F32([e]) => e.abs().into(),
+                    Scalar::AbstractInt([e]) => e.abs().into(),
+                    Scalar::I32([e]) => e.abs().into(),
```

Comment:
```suggestion
                    Scalar::I32([e]) => e.wrapping_abs().into(),
```

We should use `wrapping_abs` per spec.

Reply:
Good catch! Fixed up with e6f60f255.